plugins/modules/site_workflow_manager.py

```diff
@@ -901,17 +1532,117 @@ def delete_single_site(self, site_id, site_name):
+            self.log(
+                "Deleting floor site: {0} with ID: {1}".format(site_name, site_id), "INFO")
+            response = self.dnac._exec(
+                family="site_design",
+                function="deletes_a_floor",
+                op_modifies=True,
+                params={'id': site_id},
+            )
+            self.log(response)
+            return response
+
+        except Exception as e:
+            error_msg = "Exception occurred while deleting floor site '{0}' with site_id '{1}' due to: {2}".format(site_name, site_id, str(e))
+            self.log(error_msg, "ERROR")
+            self.status = "failed"
+            self.msg = error_msg
+
+        return self
+
+    def delete_building(self, site_name):
+        """
+        Deletes a building site by ID.
+
+        Parameters:
+            site_id (str): The ID of the building site to be deleted.
+            site_name (str): The name of the building site to be deleted.
+
+        Returns:
+            dict: The API response from the 'deletes_a_building' operation.
+        """
+        site_id = self.have.get("site_id")
+        try:
+
+            self.log(
+                "Deleting building site: {0} with ID: {1}".format(site_name, site_id), "INFO")
+
+            response = self.dnac._exec(
+                family="site_design",
+                function="deletes_a_building",
+                op_modifies=True,
+                params={'id': site_id},
+            )
+            self.log(response)
+            return response
+
+        except Exception as e:
+            error_msg = "Exception occurred while deleting building site '{0}' with site_id '{1}' due to: {2}".format(site_name, site_id, str(e))
+            self.log(error_msg, "ERROR")
+            self.status = "failed"
+            self.msg = error_msg
+        return self
+
+    def delete_area(self, site_name):
+        """
+        Deletes an area site by ID.
+
+        Parameters:
+            site_id (str): The ID of the site to be deleted.
+            site_name (str): The name of the site to be deleted.
+
+        Returns:
+            self: An instance of the class used for interacting with Cisco Catalyst Center.
+        """
+        site_id = self.have.get("site_id")
+        try:
+            self.log(
+                "Deleting area site: {0} with ID: {1}".format(site_name, site_id), "INFO")
+            response = self.dnac._exec(
+                family="site_design",
+                function="deletes_an_area",
+                op_modifies=True,
+                params={'id': site_id},
+            )
+            self.log(response)
+            return response
+
+        except Exception as e:
+            error_msg = "Exception occurred while deleting area site '{0}' with site_id '{1}' due to: {2}".format(site_name, site_id, str(e))
+            self.log(error_msg, "ERROR")
+            self.status = "failed"
+            self.msg = error_msg
+
+        return self
```

Comment:
What we are returning from this method either it's response or instance of class.

Reply:
response when the floor update is successful, and return self on failure(exception)